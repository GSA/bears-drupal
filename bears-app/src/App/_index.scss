#usagov-bears-app {
  html,
  .bears-app.form {
    background-color: #1a4480;
    height: 100%;
    min-height: 100vh;
  }

  .usa-button {
    background-color: #005ea2 !important;
    color: #fff !important;
  }

  .usa-button:hover {
    background-color: #003e6b !important;
    color: #fff !important;
  }
  /* stylelint-disable selector-class-pattern */
  .usa-button--outline {
    background-color: #fff !important;
    box-shadow: inset 0 0 0 2px #005ea2 !important;
    color: #005ea2 !important;
  }

  .usa-button--outline:hover {
    background-color: #fff !important;
    box-shadow: inset 0 0 0 2px #003e6b !important;
    color: #003e6b !important;
  }

  .usa-step-indicator__segment-label {
    color: #565c65;
  }

  .usa-step-indicator__segment--complete .usa-step-indicator__segment-label {
    color: #003e6b !important;
  }

  .usa-step-indicator__segment--current::after {
    background-color: #003e6b !important;
  }

  .usa-step-indicator__segment--complete::after {
    background-color: #003e6b !important;
  }

  .usa-hint--required {
    color: #71767a !important;
  }

  .usa-radio__input:checked + .usa-radio__label::before {
    background-color: #003e6b;
    box-shadow: 0 0 0 2px #003e6b;
  }
  /* stylelint-enable selector-class-pattern */
}

// try some usa-gov overides
.main-content {
  padding-bottom: 0 !important;
}

.usa-footer {
  padding-top: 0 !important;
}

#nav-modal {
  .usa-button {
    background-color: #005ea2 !important;
    color: #fff !important;
    width: 100%;
  }

  .usa-button:hover {
    background-color: #003e6b !important;
    color: #fff !important;
  }
}
