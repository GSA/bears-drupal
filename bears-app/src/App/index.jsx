import { useState, createContext } from 'react'
import {
  Intro,
  LifeEventSection,
  ResultsView,
  VerifySelectionsView,
  Form,
} from '../shared/components'

// data and ui content
import * as en from '../shared/locales/en/en.json'
import * as es from '../shared/locales/es/es.json'
import content from '../shared/api/mock-data/content-data'

/**
 * a functional component that renders our application.
 * @component
 */
function App() {
  // we create context state to provide translations for our two languages
  const LanguageContext = createContext({ en, es })

  // handlers
  /**
   * a function that determines the context of ui translations
   * based on a string match in the pathname of the window object
   * @function
   */
  const handleLanguage = () => {
    const string = /^\/es/
    return string.test(window.location.pathname) ? es : en
  }

  // destructure data
  const { lifeEventForm } = JSON.parse(content)
  const stepDataArray = lifeEventForm.sectionsEligibilityCriteria

  // state
  const [t] = useState(handleLanguage) // tranlations
  const [step, setStep] = useState(0) // steps indicator
  const [stepData, setStepData] = useState(stepDataArray[step]) // content
  const [verifyStep, setVerifyStep] = useState(false) // verification view
  const [viewResults, setViewResults] = useState(false) // resuts view

  return (
    <LanguageContext.Provider value={t}>
      {/* {console.log(t)} */}
      <div className="bears-app">
        {step === 0 ? (
          <Intro
            data={lifeEventForm}
            ui={t.intro}
            setStep={setStep}
            step={step}
          />
        ) : viewResults === true ? (
          <ResultsView
            ui={t.resultsView}
            handleStepBack={() => {
              setVerifyStep(false)
              setViewResults(false)
            }}
          />
        ) : verifyStep === false ? (
          <Form>
            <LifeEventSection
              step={step}
              setStep={setStep}
              data={stepDataArray}
              stepData={stepData}
              setStepData={setStepData}
              verifyStep={verifyStep}
              setVerifyStep={() => setVerifyStep(true)}
              setViewResults={() => setViewResults(true)}
              ui={t}
            />
          </Form>
        ) : (
          <VerifySelectionsView
            handleStepBack={() => {
              setVerifyStep(false)
              setViewResults(false)
            }}
            handleStepForward={() => {
              setViewResults(true)
            }}
            ui={t.verifySelectionsView}
          />
        )}
      </div>
    </LanguageContext.Provider>
  )
}

export default App
