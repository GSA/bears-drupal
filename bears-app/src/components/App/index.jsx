import { useEffect, useState } from 'react'
import { ComponentSandbox } from '../index'
import { Button } from '../shared'

/**
 * a functional component that renders our application.
 * @component
 */
function App() {
  /**
   * a boolean function to manage the guide visibility state.
   * @function
   * @param {boolean} showGuide - the state of environment
   * @return {state} returns null if not production
   */
  const [showGuide, setShowGuide] = useState(
    process.env.NODE_ENV !== 'production' ? true : null
  )

  /**
   * a fetch function to GET the data object.
   * @function
   * @param {fetch} setData - the state of environment
   * @return {state} returns data object
   */
  const [data, setData] = useState(null)

  /**
   * a boolean function to manage the visibility of the guide button.
   * @function
   * @param {boolean} showGuide - The inherited class from
   * @return {component} returns a component if not null
   */
  const handleShowGuideButton = showGuide => {
    return (
      showGuide !== null && (
        <Button
          className="usa-button--outline"
          onClick={() => setShowGuide(!showGuide)}
        >
          Component Sandbox
        </Button>
      )
    )
  }

  /**
   * a boolean function to manage the visibility of the ComponentSandbox.
   * @function
   * @param {boolean} showGuide - The inherited class from
   * @return {component} returns a component if not null
   */
  const handleShowApp = showGuide => {
    return showGuide ? <ComponentSandbox /> : <h1>App</h1>
  }

  async function GETLifeEvent(lifeEvent) {
    const response = await fetch(`/bears/api/life-event/${lifeEvent}`)
    const jsonData = await response.json()
    return jsonData
  }

  useEffect(() => {
    GETLifeEvent('retirement')
      .then(response => setData(response))
      .then(console.log(data))
    return () => {}
  }, [data])

  return (
    <div className="App">
      {handleShowGuideButton(showGuide)}
      {handleShowApp(showGuide)}
      {data?.data.map(d => JSON.stringify(d))}
    </div>
  )
}

export default App
