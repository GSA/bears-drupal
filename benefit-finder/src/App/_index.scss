@use '../shared/styles/colors/index.scss' as color;
@use '../shared/styles/mixins/_index.scss' as *;
@use '../shared/styles/sizes/_index.scss' as *;

#benefit-finder {
  html,
  .benefit-finder.form {
    font-family: 'Public Sans', sans-serif;
    background-color: color.$officer-navy;
    height: 100%;
    min-height: 100vh;
  }

  .usa-step-indicator__segment-label {
    color: color.$dark-medium-grey;
  }

  .usa-step-indicator__segment--complete .usa-step-indicator__segment-label {
    color: color.$officer-navy !important;
  }

  .usa-step-indicator__segment--current::after {
    background-color: color.$officer-navy !important;
  }

  .usa-step-indicator__segment--complete::after {
    background-color: color.$officer-navy !important;
  }

  .usa-radio__input:checked + .usa-radio__label::before {
    background-color: #005ea2 !important;
    box-shadow: 0 0 0 2px #005ea2, inset 0 0 0 2px white !important;
  }

  .a11y-sr-only {
    @include sr-only;
  }
}

// try some usa-gov overides
.main-content {
  padding-bottom: 0 !important;
}

.usa-footer {
  padding-top: 0 !important;
}

.nav-item-one,
.nav-item-two {
  &.usa-button {
    @include important-bold;

    background-color: color.$pop-blue !important;
    color: color.$white !important;

    // width: 100% !important;
    width: $button-max-width !important;
  }

  &.usa-button:hover {
    background-color: color.$officer-navy !important;
    color: color.$white !important;
  }
}
