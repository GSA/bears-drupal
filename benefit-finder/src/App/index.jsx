import { useState, createContext, useEffect } from 'react'
import * as apiCalls from '../shared/api/apiCalls'
import {
  Intro,
  LifeEventSection,
  ResultsView,
  VerifySelectionsView,
  Form,
} from '../shared/components'

import './_index.scss'

// data and ui content
import * as en from '../shared/locales/en/en.json'
import * as es from '../shared/locales/es/es.json'
// import content from '../shared/api/mock-data/current.js'

/**
 * a functional component that renders our application.
 * @component
 */
function App({ appContent }) {
  // we create context state to provide translations for our two languages
  const LanguageContext = createContext({ en, es })
  const sharedToken = 'shared'
  const windowQuery = window.location.search
  const hasQueryParams = windowQuery.includes(sharedToken)

  /**
   * lazy load our data state.
   * @function
   * @param {promise} setData - the state of environment
   * @return {state} returns null if not set
   */
  const [content, setContent] = useState(() => {
    apiCalls.GET.LifeEvent().then(response =>
      response.data ? setContent(response.data) : setContent(appContent)
    )
  })

  // set data state
  const [stepDataArray, setStepDataArray] = useState()
  const [benfitsArray, setBenefitsArray] = useState()
  const [modalOpen, setModalOpen] = useState(false)

  useEffect(() => {
    content && setBenefitsArray([...content.benefits])
    content &&
      setStepDataArray([...content.lifeEventForm.sectionsEligibilityCriteria])
  }, [content])

  // state
  const [t] = useState(apiCalls.GET.Language() === 'es' ? es : en) // tranlations
  const [step, setStep] = useState(hasQueryParams ? null : 0) // steps indicator

  const [stepData, setStepData] = useState(
    () => stepDataArray && stepDataArray[step]
  ) // content
  const [verifyStep, setVerifyStep] = useState(false) // verification view
  const [viewResults, setViewResults] = useState(hasQueryParams) // resuts view

  useEffect(() => {
    if (hasQueryParams) {
      stepDataArray &&
        apiCalls.PUT.DataFromParams(
          windowQuery,
          stepDataArray,
          setStepData,
          sharedToken
        )
      stepDataArray && setStep(stepDataArray.length)
    }
  }, [windowQuery, hasQueryParams, stepDataArray])

  return (
    content && (
      <LanguageContext.Provider value={t}>
        <div
          className={`benefit-finder ${
            step !== 0 && viewResults !== true ? 'form' : ''
          }`}
        >
          {step === 0 ? (
            <Intro
              data={content.lifeEventForm}
              ui={t.intro}
              setStep={setStep}
              step={step}
            />
          ) : viewResults === true ? (
            <ResultsView
              stepDataArray={stepDataArray}
              data={benfitsArray}
              setBenefitsArray={() => setBenefitsArray()}
              ui={t.resultsView}
              handleStepBack={() => {
                setVerifyStep(false)
                setViewResults(false)
              }}
            />
          ) : verifyStep === false ? (
            <div>
              <Form>
                <LifeEventSection
                  step={step}
                  setStep={setStep}
                  data={stepDataArray}
                  handleData={setStepDataArray}
                  stepData={stepData}
                  setStepData={setStepData}
                  verifyStep={verifyStep}
                  setVerifyStep={() => {
                    setVerifyStep(true)
                    setModalOpen(false)
                  }}
                  setViewResults={() => {
                    setViewResults(true)
                    setModalOpen(false)
                  }}
                  ui={t}
                  modalOpen={modalOpen}
                  setModalOpen={setModalOpen}
                />
              </Form>
            </div>
          ) : (
            <VerifySelectionsView
              handleStepBack={() => {
                setVerifyStep(false)
                setViewResults(false)
              }}
              handleStepForward={() => {
                setViewResults(true)
              }}
              ui={t}
              data={stepDataArray}
              step={step}
              setStep={setStep}
            />
          )}
        </div>
      </LanguageContext.Provider>
    )
  )
}

export default App
