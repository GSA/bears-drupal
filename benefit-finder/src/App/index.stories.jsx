import App from './index.jsx'
import content from '../shared/api/mock-data/current.js'

const { data } = JSON.parse(content)

export default {
  component: App,
  tags: ['autodocs'],
  args: {
    appContent: data,
  },
  parameters: {
    layout: 'fullscreen',
  },
}

export const Primary = {}
