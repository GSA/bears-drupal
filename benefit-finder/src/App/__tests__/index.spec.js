import TestRenderer from 'react-test-renderer'
import App from '../index'

describe('App', () => {
  it('renders a match to the previous snapshot', () => {
    const component = TestRenderer.create(<App />)
    expect(component.toJSON()).toMatchSnapshot()
  })
})
