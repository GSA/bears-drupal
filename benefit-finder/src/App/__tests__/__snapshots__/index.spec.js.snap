// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders a match to the previous snapshot 1`] = `null`;
